packages/dx-react-grid/src/components/table-layout.jsx

```diff
@@ -3,150 +3,68 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { findDOMNode } from 'react-dom';
+import { TemplateRenderer } from '@devexpress/dx-react-core';
 import {
-  DropTarget,
-  TemplateRenderer,
-} from '@devexpress/dx-react-core';
-import {
-  TABLE_DATA_TYPE,
-  getTableColumnGeometries,
-  getTableTargetColumnIndex,
   getAnimations,
   filterActiveAnimations,
   evalAnimations,
 } from '@devexpress/dx-grid-core';
-
 import { RowsBlockLayout } from './table-layout/rows-block-layout';
 
 const TABLE_FLEX_TYPE = 'flex';
 
+const areColumnsChanged = (prevColumns, nextColumns) => {
+  if (prevColumns.length !== nextColumns.length) return true;
+  const prevKeys = prevColumns.map(column => column.key);
+  return nextColumns.filter(column =>
```

Comment:
Can we use `Array.find` or `Array.findIndex` here? It doesn't iterate all array items.
```js
return nextColumns.find(column => prevKeys.indexOf(column.key) === -1) !== undefined;
```